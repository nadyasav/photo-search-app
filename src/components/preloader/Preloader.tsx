import React from 'react';
import { IPreloaderDivStyled, IPreloaderProps } from '../../types/customTypes';
import styles from './Preloader.module.css';
import styled from 'styled-components';

const PreloaderDiv = styled.div<IPreloaderDivStyled>`
  margin-top: ${(props) => props?.margin?.top || '0'};
  margin-right: ${(props) => props?.margin?.right || '0'};
  margin-bottom: ${(props) => props?.margin?.bottom || '0'};
  margin-left: ${(props) => props?.margin?.left || '0'};
  position: ${(props) => props?.position || 'absolute'};
`;

const Circle1 = styled.div<{ circle1: string }>`
  ${(props) => props?.circle1 || ''}
`;

const Circle2 = styled.div<{ circle2: string }>`
  ${(props) => props?.circle2 || ''}
`;

const Circle3 = styled.div<{ circle3: string }>`
  ${(props) => props?.circle3 || ''}
`;

function Preloader(props?: IPreloaderProps) {
  return (
    <div data-testid="preloader" className={styles.preloader__box}>
      <PreloaderDiv className={styles.preloader} {...props?.styled?.preloaderDiv}>
        <Circle1 data-testid="circle1" className={styles.circle1} {...props?.styled}>
          <Circle2 data-testid="circle2" className={styles.circle2} {...props?.styled}></Circle2>
          <Circle3 data-testid="circle3" className={styles.circle3} {...props?.styled}></Circle3>
        </Circle1>
      </PreloaderDiv>
    </div>
  );
}

export default React.memo(Preloader);
