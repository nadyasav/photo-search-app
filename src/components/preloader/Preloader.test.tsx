import { render } from '@testing-library/react';
import React from 'react';
import Preloader from './Preloader';

const styled = {
  preloaderDiv: {
    margin: { top: '20px', right: '20px', bottom: '20px', left: '20px' },
    position: 'sticky',
  },
  circle1: 'border-color: #ffffff;',
  circle2: 'border-color: #ffffff;',
  circle3: 'border-color: #ffffff;',
};

describe('Testing Preloader ', () => {
  it('Renders Preloader', () => {
    const preloader = render(<Preloader />);
    expect(preloader).toMatchSnapshot();
  });
  it('Renders Preloader with props', () => {
    const preloader = render(<Preloader styled={styled} />);
    expect(preloader).toMatchSnapshot();
  });
});
