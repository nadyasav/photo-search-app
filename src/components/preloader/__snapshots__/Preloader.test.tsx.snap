// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Preloader  Renders Preloader 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="preloader__box"
        data-testid="preloader"
      >
        <div
          class="sc-bcXHqe bAsrGT preloader"
        >
          <div
            class="sc-gswNZR kWSsUM circle1"
            data-testid="circle1"
          >
            <div
              class="sc-dkrFOg bdXvjK circle2"
              data-testid="circle2"
            />
            <div
              class="sc-hLBbgP fTgiIa circle3"
              data-testid="circle3"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="preloader__box"
      data-testid="preloader"
    >
      <div
        class="sc-bcXHqe bAsrGT preloader"
      >
        <div
          class="sc-gswNZR kWSsUM circle1"
          data-testid="circle1"
        >
          <div
            class="sc-dkrFOg bdXvjK circle2"
            data-testid="circle2"
          />
          <div
            class="sc-hLBbgP fTgiIa circle3"
            data-testid="circle3"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Preloader  Renders Preloader with props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="preloader__box"
        data-testid="preloader"
      >
        <div
          class="sc-bcXHqe jOkCbb preloader"
        >
          <div
            class="sc-gswNZR gDSsle circle1"
            data-testid="circle1"
          >
            <div
              class="sc-dkrFOg jQSaNk circle2"
              data-testid="circle2"
            />
            <div
              class="sc-hLBbgP iZpnOM circle3"
              data-testid="circle3"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="preloader__box"
      data-testid="preloader"
    >
      <div
        class="sc-bcXHqe jOkCbb preloader"
      >
        <div
          class="sc-gswNZR gDSsle circle1"
          data-testid="circle1"
        >
          <div
            class="sc-dkrFOg jQSaNk circle2"
            data-testid="circle2"
          />
          <div
            class="sc-hLBbgP iZpnOM circle3"
            data-testid="circle3"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
